client/src/app/site/agenda/components/list-of-speakers/list-of-speakers.component.ts

```diff
@@ -339,9 +377,26 @@ export class ListOfSpeakersComponent extends BaseViewComponent implements OnInit
      * @param speaker The speaker clicked on.
      */
     public onMarkButton(speaker: ViewSpeaker): void {
-        this.listOfSpeakersRepo
-            .markSpeaker(this.viewListOfSpeakers, speaker, !speaker.marked)
-            .then(null, this.raiseError);
+        this.listOfSpeakersRepo.markSpeaker(this.viewListOfSpeakers, speaker, !speaker.marked).catch(this.raiseError);
+    }
+
+    /**
+     * Removes the last finished speaker from the list an re-adds him on pole position
+     * Re-adding and sorting has to be done seperately right now which causes a quick jump in the list
+     */
+    public async addLastSpeakerToList(): Promise<void> {
```

Comment:
This method does 3 requests to the server. We like to avoid this due to (upcomming) eventual consistancy: Please do every action in one requests. So: A new view is needed on the server doing exactly what you want here. Please see the `agenda/views.py` for requests endpoints. For reference, look at other views.py's as well as https://www.django-rest-framework.org/api-guide/viewsets/

This method becomes obsolete.